### PR TITLE
Add a messenger share button

### PR DESCRIPTION
**Pull request by mchv**
## What does this change?

This adds a share on messenger social button.
## What is the value of this and can you measure success?

Facebook want us to test out their new messenger share option - it's currently in beta. We're trying to build a trustful relationship with them and being early adopters of their platform and new services should help with that.  

We also don't know, how much of our audience uses Facebook Messenger. We would like to use the data from this social sharing option as a proxy to answer this question.
## Screenshots

screen shot 2016-05-26 at 16 03 25

### CODE REVIEW

**Comment by cb372** [comment]
Why isn't the Pinterest button shown in your screenshot?

**Comment by paperboyo** [comment]
https://youtu.be/VfcBxu101qM?t=2

Maybe we could hide them all behind one button?

**Comment by ScottPainterGNM** [comment]
@cb372 I think we only show pinterest on galleries and whatsapp is mobile breakpoint only, I'm guessing the latter would be appropriate for messenger as well?

**Comment by LATaylor-guardian** [comment]
@mchv Can we see how many people are clicking this button as soon as this change happens without further changes elsewhere?

**Comment by mchv** [comment]
@LATaylor-guardian  I think I need first to make a chance to the fastly cache to redirect `sme` to a campaign code. 
@cb372 I have been playing with removing `pinterest` button, but I think it is not necessary a first step.

**Comment by mchv** [comment]
@LATaylor-guardian change was in frontend and I just made it 😃

File: static/src/stylesheets/module/_social.scss [hunk 1]
```diff
@@ -159,6 +165,13 @@ category: Common
     }
 }
 
+.social-icon--messenger {
+    background-color: $social-messenger;
+    &:hover {
+        background-color: darken($social-messenger, 10%);
```
Comment (gtrufitt) [inline static/src/stylesheets/module/_social.scss]: Is this, erm, official Facebook brand hover colour..? Also, you might want to add `:focus` and `:active` states here.

File: common/app/model/ShareLinks.scala [hunk 2]
```diff
@@ -28,6 +28,14 @@ object Facebook extends SharePlatform {
   override val css = "facebook"
   override val userMessage = "Share on Facebook"
 }
+
+object Messenger extends SharePlatform {
+  override val campaign = Some("sme")
+  override val text = "Messenger"
+  override val css = "messenger"
+  override val userMessage = "Share on Messenger"
```
Comment (gtrufitt) [inline common/app/model/ShareLinks.scala]: (minor) 'Facebook Messenger' rather than just 'messenger'?

Comment (mchv) [inline common/app/model/ShareLinks.scala]: The app is called `Messenger` (see https://www.messenger.com/about)

File: static/src/stylesheets/module/_social.scss (hunk 1 above)
Comment (mchv) [inline static/src/stylesheets/module/_social.scss]: `10%` is what we have for `twitter`, `whatsapp`, `gplus`, so I don't think we should be more specific for messenger here.

**Comment by johnduffell** [comment]
ok it's great to share, but surely we can't just keep adding one more share button? I'm sure the optimal number of share buttons must be around 2.

File: static/src/stylesheets/module/_social.scss (hunk 1 above)
Comment (mchv) [inline static/src/stylesheets/module/_social.scss]: Thanks for `:focus` and `:active` states but I think, if we add them, we should add them for all social icons, and so in a separate pr.

**Comment by mchv** [comment]
@johnduffell Yep. We are [monitoring the shares](https://docs.google.com/spreadsheets/d/1HNGgHdfYKPUhhzwt3uqVkLFFuXePSMRcML1RlsSSHX0/edit?ts=5729b71e#gid=1074393378) and we will in the future remove the sharing that are less used. Please note as well that like `whatsapp`, `messenger` only show on mobile 😃

File: static/src/stylesheets/module/_social.scss (hunk 1 above)
Comment (gtrufitt) [inline static/src/stylesheets/module/_social.scss]: Fair enough

File: common/app/model/ShareLinks.scala (hunk 2 above)
Comment (gtrufitt) [inline common/app/model/ShareLinks.scala]: Yeah, noticed that after, trust the big F to TM such a generic name :D

**Comment by johnduffell** [comment]
this is making the amp pages invalid so they're not appearing in search any more

**Pull request merged**